<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>StrokeDefinitionDialog</class>
 <widget class="QDialog" name="StrokeDefinitionDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>448</width>
    <height>562</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>448</width>
    <height>539</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Skeleton Stroke Definition</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../../rsc/espina.qrc">
    <normaloff>:/espina/espina.svg</normaloff>:/espina/espina.svg</iconset>
  </property>
  <property name="toolTip">
   <string>Define skeleton stroke types and properties</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0">
     <item>
      <widget class="QListWidget" name="m_list"/>
     </item>
     <item>
      <layout class="QVBoxLayout" name="m_propertiesLayout">
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <property name="topMargin">
          <number>9</number>
         </property>
         <item row="0" column="1">
          <widget class="QLineEdit" name="m_name">
           <property name="acceptDrops">
            <bool>false</bool>
           </property>
           <property name="toolTip">
            <string>Enter stroke type identifier</string>
           </property>
           <property name="statusTip">
            <string>Enter stroke type identifier</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_2">
           <property name="toolTip">
            <string>Stroke type identifier</string>
           </property>
           <property name="statusTip">
            <string>Stroke type identifier</string>
           </property>
           <property name="text">
            <string>Name:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_3">
           <property name="statusTip">
            <string>Stroke type color</string>
           </property>
           <property name="whatsThis">
            <string>Stroke type color</string>
           </property>
           <property name="text">
            <string>Color:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="ESPINA::HueSelector" name="m_hueWidget" native="true">
           <property name="toolTip">
            <string>Define stroke color</string>
           </property>
           <property name="statusTip">
            <string>Define stroke color</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Type:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QComboBox" name="m_typeCombo">
           <property name="toolTip">
            <string>Define stroke line type</string>
           </property>
           <property name="statusTip">
            <string>Define stroke line type</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QCheckBox" name="m_useCategoryColor">
           <property name="text">
            <string>Use category color</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="m_validMeasure">
         <property name="toolTip">
          <string>Check if the measure is to be used in skeleton computations</string>
         </property>
         <property name="text">
          <string>Use measure</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="label_4">
     <property name="text">
      <string>Strokes that use a color other than that of the category will be displayed in that color regardless of the color engine used.</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignJustify|Qt::AlignVCenter</set>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_5">
     <property name="text">
      <string>If the 'Use measure' option is not checked then the stroke won't be taken into consideration when computing the skeleton length.</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignJustify|Qt::AlignVCenter</set>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_6">
     <property name="text">
      <string>Warning: deleting a stroke will also delete all edges in the skeleton related to that stroke.</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignJustify|Qt::AlignVCenter</set>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QPushButton" name="m_addButton">
       <property name="focusPolicy">
        <enum>Qt::TabFocus</enum>
       </property>
       <property name="contextMenuPolicy">
        <enum>Qt::NoContextMenu</enum>
       </property>
       <property name="text">
        <string>Add</string>
       </property>
       <property name="autoDefault">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="m_removeButton">
       <property name="focusPolicy">
        <enum>Qt::TabFocus</enum>
       </property>
       <property name="contextMenuPolicy">
        <enum>Qt::NoContextMenu</enum>
       </property>
       <property name="text">
        <string>Remove</string>
       </property>
       <property name="autoDefault">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="m_closeButton">
       <property name="contextMenuPolicy">
        <enum>Qt::NoContextMenu</enum>
       </property>
       <property name="text">
        <string>Close</string>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ESPINA::HueSelector</class>
   <extends>QWidget</extends>
   <header location="global">GUI/Widgets/HueSelector.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../rsc/espina.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>m_closeButton</sender>
   <signal>clicked()</signal>
   <receiver>StrokeDefinitionDialog</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>371</x>
     <y>403</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
